internal/gitpartitionsync/producer/tar.go

```diff
@@ -65,15 +64,18 @@ func (g *GitPartitionSyncProducer) tarRepos(repoPath string, sync syncConfig) (s
 		if err != nil {
 			return err
 		}
+		// Closing a healthy file twice yields a syscall.EINVAL
+		// error which is safe to discard in this case.
+		defer func() { _ = f.Close() }()
 
 		// copy file data into tar writer
 		if _, err := io.Copy(tw, f); err != nil {
 			return err
 		}
 
-		// manually close here after each file operation; defering would cause each file close
```

Comment:
**To the reviewer:**

This is fine for this specific code block, as the filepath.Walk() function uses a callback, and since defer only fires when a function terminates and goes out of scope - in other words, any given defer scope is to a function that encloses it. This is, of course, different for loops and iterators where caution has to be employed.